app/model/ThriftUtil.scala

```diff
@@ -67,19 +67,18 @@ object ThriftUtil {
     }
     val duration = params.get("duration").map(_.head.toLong)
     val source = params.get("source").map(_.head)
+    val posterUrl = params.get("posterUrl").map(_.head).map(parseId).flatMap(_.right.toOption)
```

Comment:
do we need to use `parseId` here? I am assuming that all we want to do is check that it is a URL, and that it will never need to be distinguished as a youtube id? In that case maybe we can just apply `Url` directly.